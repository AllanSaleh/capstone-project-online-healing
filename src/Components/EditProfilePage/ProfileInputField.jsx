import React from 'react';

const ProfileInputField = ({ title, id, type, icon }) => (
  <div className="flex items-center my-6">
    <label htmlFor={id} className="text-subtitle capitalize inline-block w-1/3">
      {title}
    </label>
    <div className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg pl-4">
      <input type={type} id={id} name={id} className="w-11/12 h-full outline-none py-3" />
      {icon && (
        <button type="button">
          <img src={icon} alt="profileIcon" />
        </button>
      )}
    </div>
  </div>
);

export default ProfileInputField;
