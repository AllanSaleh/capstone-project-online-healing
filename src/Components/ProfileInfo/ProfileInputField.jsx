import React from 'react';

const ProfileInputField = ({ title, id, type, icon }) => (
  <div className="flex items-center my-6">
    <label htmlFor={id} className="text-subtitle capitalize inline-block w-1/3">{title}</label>
    <div className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg pl-4">
      <input type={type} id={id} name={id} className="w-11/12 h-full outline-none py-3" />
      {icon && (
        <button>
          <img src={icon} alt="icon" />
        </button>
      )}
    </div>
  </div>
);

export default ProfileInputField;
