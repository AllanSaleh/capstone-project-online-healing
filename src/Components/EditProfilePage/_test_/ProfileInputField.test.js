import * as React from 'react';
import * as renderer from 'react-test-renderer';
import ProfileInputField from '../ProfileInputField';

it('Profile Input Field renders correctly', () => {
  const tree = renderer.create(<ProfileInputField />).toJSON();
  expect(tree).toMatchSnapshot();
});
