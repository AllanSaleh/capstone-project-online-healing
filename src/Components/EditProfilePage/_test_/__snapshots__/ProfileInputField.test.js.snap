// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex items-center my-6"
>
  <label
    className="text-subtitle capitalize inline-block w-1/3"
  />
  <div
    className="w-2/3 inline-flex items-center border border-black border-opacity-20 rounded-lg pl-4"
  >
    <input
      className="w-11/12 h-full outline-none py-3"
    />
  </div>
</div>
`;
